arch/arm/src/arm/arm_initialstate.c

```diff
@@ -75,8 +75,10 @@ void up_initial_state(struct tcb_s *tcb)
 
   if (tcb->pid == 0)
     {
-      up_use_stack(tcb, (void *)(g_idle_topstack -
-        CONFIG_IDLETHREAD_STACKSIZE), CONFIG_IDLETHREAD_STACKSIZE);
+      tcb->stack_alloc_ptr = (void *)(g_idle_topstack -
+                                      CONFIG_IDLETHREAD_STACKSIZE);
+      tcb->adj_stack_ptr   = (void *)g_idle_topstack;
+      tcb->adj_stack_size  = CONFIG_IDLETHREAD_STACKSIZE;
     }
```

Comment:
@xiaoxiang781216 

I think we should add such special logic for pid=0 case to up_use_stack().
It would reduce maintenance costs.

What do you think?

Reply:
We agreed that idle thread colorization should be done in assembly code from https://github.com/apache/incubator-nuttx/pull/1369, so up_use_stack isn't used to color the idle thread stack anymore.
BTW, up_use_stack and idle thread tcb is seldom referenced in the code base and the check is introduced in https://github.com/apache/incubator-nuttx/pull/1369 too.

Reply:
OK.